pkg/controllers/cluster/import.go

```diff
@@ -139,6 +140,21 @@ func (i *importHandler) deleteOldAgent(cluster *fleet.Cluster, kc kubernetes.Int
 		}
 	}
 
+	if i.systemNamespace != config.DefaultNamespace {
+		// Attempt to delete the agent in the default namespace if it has changed.
+		if err := kc.AppsV1().Deployments(config.DefaultNamespace).Delete(i.ctx, config.AgentConfigName, metav1.DeleteOptions{}); err != nil {
+			return err
+		}
+		apply, err := apply.NewForConfig(restConfig)
+		if err != nil {
+			return err
+		}
+		// Using apply with nil objects to delete objects in the default namespace.
+		if err := apply.WithSetID(deployer.GetSetID(config.AgentConfigName, config.DefaultNamespace)).ApplyObjects(nil); err != nil {
+			return err
+		}
```

Comment:
We might even want to remove this if we want to leave everything else behind. thoughts?

Reply:
It is better to clean up other resources given the fact that we are already cleaning fleet-agent deployment. The apply is not going to work most likely. We can try deleting it manually.